docs/the-new-architecture/pillars-turbomodule.md

```diff
@@ -3,16 +3,699 @@ id: pillars-turbomodules
+
+As first step for both iOS and Android, this guide shows how to execute manually the scripts used by **CodeGen** to generate the required code. Further information on **CodeGen** can be found [here](/docs/pillars-codegen.md)
+
+:::caution
+The code generated by the **CodeGen** in this step should not be committed to the versioning system. React Native apps are able to generate the code when the app is built. This allows to avoid any ABI incompatibility and to ensure that a consistent version of the **CodeGen** is used.
+:::
+
+### iOS
+
+#### Generate the code - iOS
+
+To run Codegen for the iOS platform, we need to open a terminal and run the following command:
+
+```sh title="Running CodeGen for iOS"
+cd MyApp
+yarn add ../RTNCalculator
+cd ..
+node MyApp/node_modules/react-native/scripts/generate-artifacts.js \
+  --path MyApp/ \
+  --outputPath RTNCalculator/generated/
+```
+
+This script first adds the `RTNCalculator` module to the app with `yarn add`. Then, it invokes Codegen via the `generate-artifacts.js` script.
+
+The `--path` option specifies the path to the app, while the `--outputPath` option tells the script where to output the generated code.
+
+The output of this process is the following folder structure:
+
+```sh
+generated
+└── build
+    └── generated
+        └── ios
+            ├── FBReactNativeSpec
+            │   ├── FBReactNativeSpec-generated.mm
+            │   └── FBReactNativeSpec.h
+            ├── RCTThirdPartyFabricComponentsProvider.h
+            ├── RCTThirdPartyFabricComponentsProvider.mm
+            ├── RTNCalculatorSpec
+            │   ├── RTNCalculatorSpec-generated.mm
+            │   └── RTNCalculatorSpec.h
+            └── react
+                └── renderer
+                    └── components
+                        └── rncore
+                            ├── ComponentDescriptors.h
+                            ├── EventEmitters.cpp
+                            ├── EventEmitters.h
+                            ├── Props.cpp
+                            ├── Props.h
+                            ├── RCTComponentViewHelpers.h
+                            ├── ShadowNodes.cpp
+                            └── ShadowNodes.h
+```
+
+The relevant path for the TurboModule interface is `generated/build/generated/ios/RTNCalculatorSpec`.
+
+See the [CodeGen](./pillars-codegen) section for further details on the generated files.
+
+#### Write the Native iOS Code
+
+Now add the Native code for your TurboModule. Create two files in the `RTNCalculator/ios` folder:
+
+1. The `RTNCalculator.h`, a header file for the module.
+2. The `RTNCalculator.mm`, the implementation of the module.
+
+##### RTNCalculator.h
+
+```objc title="RTNCalculator.h"
+#import <React/RCTBridgeModule.h>
+
+@interface RTNCalculator : NSObject <RCTBridgeModule>
```

Comment:
The corresponding file in the Fabric guide also includes `NS_ASSUME_NONNULL_BEGIN` in this file, but the RNNewArch example doesn't. Wasn't sure if that was important to include?

Reply:
addressed